pkg/agents/AgentTypes/agent1/agent1.go

```diff
@@ -8,44 +8,25 @@ import (
 type Agent1 struct {
 	*baseAgent.BaseAgent
 	age int
-	//Energy int
+	
 }
 
-// func ReduceEnergy(a *Agent1) {
-// 	a.Energy -= 10
-// }
+
 
 func (a1 *Agent1) Activity() {
 	fmt.Println("Agent1's Activity")
-	a1.BaseAgent.Activity()
 	fmt.Printf("age: %d\n", a1.age)
-
-	// fmt.Printf("id: %s\n", a1.GetID())
-	// a1.SetName("A1")
-	// fmt.Printf("name: %s\n", a1.GetName())
-	// fmt.Printf("__________________________\n")
+	a1.BaseAgent.Activity()
+	
 }
 
-func GetAgent() *Agent1 { //why this function returns the interface?
-	ag := &Agent1{
-		BaseAgent: baseAgent.GetAgent(),
-		age:       0,
+func GetAgent(name string) *Agent1 { 
+	return &Agent1{
+		BaseAgent: baseAgent.NewAgent(name),
+		age:       0, 
```

Comment:
very close. we know for a fact that this will be "A1", so can pass it directly, instead of requiring the `name` argument